crates/pop-cli/src/commands/up/parachain.rs

```diff
@@ -0,0 +1,415 @@
+	#[arg(short, long)]
+	pub(crate) id: Option<u32>,
+	/// Path to the genesis state file. If not specified, it will be generated.
+	#[arg(short = 'G', long = "genesis-state")]
+	pub(crate) genesis_state: Option<PathBuf>,
+	/// Path to the genesis code file.  If not specified, it will be generated.
+	#[arg(short = 'C', long = "genesis-code")]
+	pub(crate) genesis_code: Option<PathBuf>,
+	/// Websocket endpoint of the relay chain.
+	#[arg(long)]
+	pub(crate) relay_url: Option<Url>,
+}
+
+impl UpChainCommand {
+	/// Executes the command.
+	pub(crate) async fn execute(self, cli: &mut impl Cli) -> Result<()> {
+		cli.intro("Deploy a chain")?;
+		let chain_config = match self.prepare_chain_for_registration(cli).await {
+			Ok(chain) => chain,
+			Err(e) => {
+				cli.outro_cancel(format!("{}", e))?;
+				return Ok(());
+			},
+		};
+		match chain_config.register_parachain(cli).await {
+			Ok(_) => cli.success("Chain deployed successfully")?,
+			Err(e) => cli.outro_cancel(format!("{}", e))?,
+		}
+		Ok(())
+	}
+
+	// Prepares the chain for registration by setting up its configuration.
+	async fn prepare_chain_for_registration(self, cli: &mut impl Cli) -> Result<UpChain> {
+		let chain = self.configure_chain(cli).await?;
+		let para_id = self.resolve_parachain_id(&chain, cli).await?;
+		let (genesis_state, genesis_code) = self.resolve_genesis_files(para_id, cli).await?;
+		Ok(UpChain { id: para_id, genesis_state, genesis_code, chain })
+	}
+
+	// Configures the chain by resolving the URL and fetching its metadata.
+	async fn configure_chain(&self, cli: &mut impl Cli) -> Result<Chain> {
+		// Resolve url.
+		let url = match &self.relay_url {
+			Some(url) => url.clone(),
+			None => {
+				// Prompt for url.
+				let url: String = cli
+					.input("Enter the relay chain node URL to deploy your parachain")
+					.default_input(DEFAULT_URL)
+					.interact()?;
+				Url::parse(&url)?
+			},
+		};
+
+		// Parse metadata from chain url.
+		let client = set_up_client(url.as_str()).await?;
+		let pallets = parse_chain_metadata(&client).map_err(|e| {
+			anyhow!(format!("Unable to fetch the chain metadata: {}", e.to_string()))
+		})?;
+		Ok(Chain { url, client, pallets })
+	}
+	// Resolves the parachain ID, reserving a new one if necessary.
+	async fn resolve_parachain_id(&self, chain: &Chain, cli: &mut impl Cli) -> Result<u32> {
+		match self.id {
+			Some(id) => Ok(id),
+			None => {
+				cli.info("Reserving a parachain ID")?;
+				reserve_para_id(chain, cli).await
+			},
+		}
+	}
+	// Resolves the genesis state and code files, generating them if necessary.
+	async fn resolve_genesis_files(
+		&self,
+		para_id: u32,
+		cli: &mut impl Cli,
+	) -> Result<(PathBuf, PathBuf)> {
+		match (self.genesis_state.clone(), self.genesis_code.clone()) {
+			(Some(state), Some(code)) => Ok((state, code)),
+			_ => {
+				cli.info("Generating the chain spec for your parachain")?;
+				generate_spec_files(para_id, self.path.clone(), cli).await
```

Comment:
```suggestion
				generate_spec_files(para_id, &self.path, cli).await
```

By updating the parameter `path` of `generate_spec_files` (line 173) to `path: &Option<PathBuf>`, we can avoid cloning here.

Reply:
Good point. I refactored to `Option<&Path>` and `self.path.as_deref()` as we have in other parts of the codebase: https://github.com/r0gue-io/pop-cli/blob/main/crates/pop-cli/src/commands/build/contract.rs#L28

https://github.com/r0gue-io/pop-cli/pull/410/commits/2548d0f1aeff9b6eab13ff550fbd4af86fe9fde1